### DISCUSSION TITLE
How to debug bundler cache (cache seems to be growing very large)

### DISCUSSION THREAD

**Discussion by Fryguy** [ruby/setup-ruby#573] [Q&A]
Our bundler cache seems to continually grow very large (right now it's at 8GB) and I'm trying to debug why. 

Is there a way to manually download the cache file so I can see what's inside?

---

My gut is telling that it's because we use git-based gems, and it's probably not "cleaning" the old git histories, but I want to prove it first before opening a ticket

**Comment by MSP-Greg** [comment]
> Is there a way to manually download the cache file so I can see what's inside?

I don't believe there is.  The cache files are placed within `vendor/bundle` in the repo.  You could add some Actions steps to log the contents...

> Our bundler cache seems to continually grow very large (right now it's at 8GB)

Is that the total, or the size of a single cache?  Note that `bundle clean` is run after `bundle install`

I don't recall where I saw the info, but I think GHA limits caches to a total of 10GB, it will then flush the older caches to stay within that limit.  Not.sure.

**Reply by Fryguy** [reply]
That's the size of a single cache.

> Note that bundle clean is run after bundle install

My current theory is that `bundle clean` doesn't clean up any old git repositories references, but I need to prove that first.

**Reply by Fryguy** [reply]
Here's a screenshot of the caches (not sure if this is publicly visible hence the screenshot)

![Workflow_runs_·_ManageIQ_manageiq-providers-ansible_tower](https://github.com/ruby/setup-ruby/assets/52120/6c2121f4-c663-4f22-b181-6cb0ad7f78f0)

I'd roughly expect closer to ~600MB which is what I see locally after doing a `bundle clean --force`, and you can see that value in some of the other caches.

**Comment by eregon** [comment]
I think you should be able to see caches and download them at https://github.com/owner/repo/actions/caches

**Reply by eregon** [reply]
Ah that doesn't allow to download it, unfortunate.
But you can add https://github.com/actions/upload-artifact or so to upload the vendor/bundle.

**Reply by Fryguy** [reply]
Oh that's an interesting approach.  Is there a limit to the size of the uploaded artifact?

(Asking because I didn't see anything in https://github.com/actions/upload-artifact?tab=readme-ov-file#limitations, but I'd be surprised if there wasn't _some_ limitation)
